mlflow/utils/requirements_utils.py

```diff
@@ -271,22 +269,56 @@ def _capture_imported_modules(model_uri, flavor):
         # resolution in a subprocess based on PATH entries.
         main_env["PATH"] = "/usr/sbin:/sbin:" + main_env["PATH"]
 
-        _run_command(
-            [
-                sys.executable,
-                _capture_modules.__file__,
-                "--model-path",
-                local_model_path,
-                "--flavor",
-                flavor,
-                "--output-file",
-                output_file,
-                "--sys-path",
-                json.dumps(sys.path),
-            ],
-            timeout_seconds=process_timeout,
-            env=main_env,
-        )
+        if flavor == mlflow.transformers.FLAVOR_NAME:
+            # Lazily import `_capture_transformers_module` here to avoid circular imports.
+            from mlflow.utils import _capture_transformers_modules
+
+            for module_to_throw in ["tensorflow", "torch", ""]:
+                try:
+                    _run_command(
+                        [
+                            sys.executable,
+                            _capture_transformers_modules.__file__,
+                            "--model-path",
+                            local_model_path,
+                            "--flavor",
+                            flavor,
+                            "--output-file",
+                            output_file,
+                            "--sys-path",
+                            json.dumps(sys.path),
+                            "--module-to-throw",
+                            module_to_throw,
+                        ],
+                        timeout_seconds=process_timeout,
+                        env=main_env,
+                    )
+                    break
+                except MlflowException:
+                    if module_to_throw == "":
+                        raise
+
+        else:
+            # Lazily import `_capture_module` here to avoid circular imports.
+            from mlflow.utils import _capture_modules
```

Comment:
```suggestion
        if flavor == mlflow.transformers.FLAVOR_NAME:
            # Lazily import `_capture_transformers_module` here to avoid circular imports.
            from mlflow.utils import _capture_transformers_modules

            for module_to_throw in ["tensorflow", "torch"]:
                try:
                    _run_command(
                        [
                            sys.executable,
                            _capture_transformers_modules.__file__,
                            "--model-path",
                            local_model_path,
                            "--flavor",
                            flavor,
                            "--output-file",
                            output_file,
                            "--sys-path",
                            json.dumps(sys.path),
                            "--module-to-throw",
                            module_to_throw,
                        ],
                        timeout_seconds=process_timeout,
                        env=main_env,
                    )
                    return
                except MlflowException:
                    pass

        # Lazily import `_capture_module` here to avoid circular imports.
        from mlflow.utils import _capture_modules
```

Can we remove `module_to_throw=""`?

Reply:
Ok